hamilton/plugins/h_dlt.py

```diff
@@ -0,0 +1,61 @@
+import logging
+from typing import Any, Dict, List, Literal, Optional, Union
+
+import dlt
+
+from hamilton import htypes
+from hamilton.graph_types import HamiltonGraph
+from hamilton.lifecycle import GraphExecutionHook
+from hamilton.plugins.dlt_extensions import DltSaver
+
+logger = logging.getLogger(__name__)
+
+
+class DltAdapter(GraphExecutionHook):
```

Comment:
I'm not seeing the value of this -- we really should have it part of the graph. If not, they should be totally separate, but this is burying critical side-effects pretty deep down...

Reply:
I think it's an interesting alternative -- definitely implicit. But something to get feedback on.

🤔  we could also model this as a result builder.